cloudflared/config.yaml

```diff
@@ -21,10 +21,20 @@ map:
 options:
   external_hostname: null
   tunnel_name: "homeassistant"
+  strict_ssl: false
+  generate_cert:
+    enabled: false
+    certfile: fullchain.pem
+    keyfile: privkey.pem
   additional_hosts: []
 schema:
   external_hostname: str
   tunnel_name: str
+  strict_ssl: bool
```

Comment:
I think we should make all those parameters optional by adding the `schema` part only (like `disableChunkedEncoding`)

Reply:
Well. The benefit is:

- User does not have even have think about it in the configuration page

The caveat is:

- User will need to open the DOCS to check what is the default value, which otherwise would be on front the configuration itself

So, as you both seem to agree that it should be transparent/optional, I will do it.

Reply:
I think we should set generate certificate to 1 anyways as a default, so that the average user will just go with it and have the SSL support as well. Therefore, it would be a pretty uncommon edge case, if you do not want to create the certificate.
If you already have certificate files in SSL, it will print an error on start-up and point you to the respecting solution / configuration option to disable for whatever reason.

What do you think?

Reply:
> I think we should set generate certificate to 1 anyways as a default, so that the average user will just go with it and have the SSL support as well.

Would that mean generating the certificates as default behavior and setting the ingress settings to noTLSVerify=false and originServerName=external_hostname? This would imply that every user who uses the add-on also configures all services with the certificate by default?

Reply:
Yes and no. So setting everything you described yes, but only for the Ingress rule of HomeAssistant. For the additional hosts, we would have the separate flag to decide.